classes/migration/install/MetricsMigration.inc.php

```diff
@@ -0,0 +1,239 @@
+<?php
+
+/**
+ * @file classes/migration/install/MetricsMigration.inc.php
+ *
+ * Copyright (c) 2014-2021 Simon Fraser University
+ * Copyright (c) 2000-2021 John Willinsky
+ * Distributed under the GNU GPL v3. For full terms see the file docs/COPYING.
+ *
+ * @class MetricsMigration
+ * @brief Describe database table structures.
+ */
+
+namespace APP\migration\install;
+
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema as Schema;
+
+class MetricsMigration extends \PKP\migration\Migration
+{
+    /**
+     * Run the migrations.
+     * This migration file is used during upgrades. If this schema changes, the upgrade scripts should be reviewed manually before a merging.
+     */
+    public function up(): void
+    {
+        Schema::create('metrics_context', function (Blueprint $table) {
+            $table->string('load_id', 255);
+            $table->bigInteger('context_id');
+            $table->date('date');
+            $table->integer('metric');
+            $table->foreign('context_id')->references('journal_id')->on('journals');
+            $table->index(['load_id'], 'metrics_context_load_id');
+            $table->index(['context_id'], 'metrics_context_context_id');
+        });
+        Schema::create('metrics_submission', function (Blueprint $table) {
+            $table->string('load_id', 255);
+            $table->bigInteger('context_id');
+            $table->bigInteger('submission_id');
+            $table->bigInteger('representation_id')->nullable();
+            $table->bigInteger('submission_file_id')->unsigned()->nullable();
+            $table->bigInteger('file_type')->nullable();
+            $table->bigInteger('assoc_type');
+            $table->date('date');
+            $table->integer('metric');
+            $table->foreign('context_id')->references('journal_id')->on('journals');
+            $table->foreign('submission_id')->references('submission_id')->on('submissions');
+            $table->foreign('representation_id')->references('galley_id')->on('publication_galleys');
+            $table->foreign('submission_file_id')->references('submission_file_id')->on('submission_files');
```

Comment:
I haven't done it yet, but I thought that when an object is deleted, the stats for that object should be deleted too. 
Maybe the representation_id could be a problem: In OJS and OPS the galley and file have relationship 1 to 1, right? So that would be OK... But I have to think about it and above OMP again... 
:thinking: